### PR TITLE
Multi arch builds

### PR DESCRIPTION
**Pull request by stevesloka**
This is a followup to #46 to enable multi client builds. 

//cc @ncdc

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -42,7 +42,13 @@ local: $(BINARIES)
 
 $(BINARIES):
 	mkdir -p $(BIN_DIR)
-	$(BUILDCMD) -o $(BIN_DIR)/$@ $(GOTARGET)/cmd/$@
+
+	mkdir -p $(BIN_DIR)/amd64
```
Comment (ncdc) [inline Makefile]: These should probably be in a different target, `cross-build` or something like that

Comment (ncdc) [inline Makefile]: Maybe we should switch to something like https://github.com/thockin/go-build-template.

**Comment by ncdc** [comment]
test this please

Let's see how long this takes to build...

**Comment by stevesloka** [comment]
My mac is building the arm docker image fine, but on my linux box, it's erroring. Let me poke around with this a bit more.

**Comment by ncdc** [comment]
It built on Jenkins, going from 2m20s to 5m13s.

**Comment by stevesloka** [comment]
@ncdc I did a revamp following Tim's example repo. Not sure how the licensing works now with that. Please take a look and let me know what you think. 

I also tested this out on my Pi cluster. I added examples for that so I could deploy, but could pick those out into another PR if that makes sense. I just kept them as I worked to test building for that architecture.

**Comment by ncdc** [comment]
@stevesloka I'm afk today but I will try to review this tomorrow. Thanks for your patience!

**Comment by ncdc** [comment]
@stevesloka if you're still interested in pursuing this, I'd like to go about this as a series of steps:

1. 1 PR to enable multi-arch builds using a `make cross-build` (or similarly named) target that builds all arches. This would not be part of the default make target, so most people building (including our CI) would only be building 1 arch by default. We'll use this target to produce precompiled binaries that we'll attach to future releases here on GitHub.
1. 1 PR to do multi-arch images. Long-term, I think using the multi-arch manifest lists would be the ideal approach, but it doesn't look like GCR supports that feature yet. In the meantime, we could use ksonnet or even create a minimal installer that lets you specify the image and/or arch you want.

Do you think you'd have time to split this up?

**Comment by stevesloka** [comment]
@ncdc Yup happy to help! I'll have some time to start to look at this next week.

**Comment by ncdc** [comment]
Great, thanks!

**Comment by ncdc** [comment]
Superseded by #148

**Pull request closed without merging**